.github/workflows/sonarqube.yml

```
@@ -46,8 +44,48 @@ jobs:
           cd ../../
           # The Vitest coverage config is in your vite.config.mts
 
-      - name: SonarQube Scan
-        uses: SonarSource/sonarqube-scan-action@bfd4e558cda28cda6b5defafb9232d191be8c203
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+      - name: Prepare SonarQube scan info
+        id: sonarqube-prep
+        # We'll collect info about the PR without running the actual scan
+        run: |
```

actionlint (1.7.4)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)